### PR TITLE
[Merged by Bors] - Update PoST dependency

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Integrates the fixed post into go-spacemesh

## Changes
Update PoST dependency and shared libraries. Merge after

- [x] https://github.com/spacemeshos/poet/pull/141
- [x] https://github.com/spacemeshos/post/pull/83

## Test Plan
Existing tests still pass.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by countvonzero — APPROVED** [review]

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

File: Makefile-gpu.Inc [hunk 1]
```diff
@@ -106,19 +106,3 @@ print-test-targets:
 	do echo $$i; \
 	done
 .PHONY: $(SUBDIRS_ALL) $(SUBDIRS_LVL2) $(SUBDIRS_ONLY) print-test-targets
-
-go-env: get-gpu-setup
-	go env -w CGO_LDFLAGS="$(CGO_LDFLAGS)"
-.PHONY: go-env
-
-go-env-test: get-gpu-setup
```
Comment (countvonzero) [inline Makefile-gpu.Inc:114]: i think this broke the configuration file in .run directory @fasmat

**Pull request closed without merging**